sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -1245,6 +1245,12 @@ export class DataLakeFileClient extends DataLakePathClient {
         pathHttpHeaders: options.pathHttpHeaders,
         tracingOptions: { ...options!.tracingOptions, spanOptions }
       });
+      // append() with empty data would return error, so do not continue
+      if (size === 0) {
+        return createRes;
+      } else {
+        await createRes;
```

Comment:
We need `await` for the tracing span to work correctly.

Reply:
Really? I don't know about that. What's the behavior and why?

Changed to 
```
return await createRes;
```

Reply:
Without await, the code in the finally block will be executed before the request finished and ends  the tracing span. We don't want this behavior.